### ISSUE TITLE
Tibber websocket API deprecated and offline

### ISSUE THREAD

**Issue by hlinden** [terjesannum/tibber-exporter#14]
Tibber has announce a while ago that they would take the old API offline:

https://developer.tibber.com/docs/overview#breaking-websocket-change

Seems that today they finally did so, at least the Exporter doesn't work anymore for me.

```
Jan 08 17:26:35 vpngateway systemd[1]: Started tibber-exporter.service - Tibber Prometheus Exporter.
Jan 08 17:26:35 vpngateway tibber-exporter[2336655]: 2024/01/08 17:26:35 Starting tibber-exporter/3.6.3 (https://github.com/terjesannum/tibber-exporter)
Jan 08 17:26:40 vpngateway tibber-exporter[2336655]: 2024/01/08 17:26:40 Websocket url: wss://websocket-api.tibber.com/v1-beta/gql/subscriptions
Jan 08 17:26:40 vpngateway tibber-exporter[2336655]: 2024/01/08 17:26:40 Found home: xxx
Jan 08 17:26:40 vpngateway tibber-exporter[2336655]: 2024/01/08 17:26:40 Starting monitoring of home: xxx
Jan 08 17:26:40 vpngateway tibber-exporter[2336655]: 2024/01/08 17:26:40 Current subscription: xxx
Jan 08 17:26:40 vpngateway tibber-exporter[2336655]: 2024/01/08 17:26:40 Creating home collector for home xxx
Jan 08 17:26:40 vpngateway tibber-exporter[2336655]: 2024/01/08 17:26:40 Realtime consumption enabled for xxx: false
Jan 08 17:26:40 vpngateway tibber-exporter[2336655]: 2024/01/08 17:26:40 Starting live measurements monitoring of home xxx
Jan 08 17:26:40 vpngateway tibber-exporter[2336655]: 2024/01/08 17:26:40 Live feed timeout: 1 minute
Jan 08 17:26:40 vpngateway tibber-exporter[2336655]: 2024/01/08 17:26:40 Updating prices for xxx
Jan 08 17:26:40 vpngateway tibber-exporter[2336655]: 2024/01/08 17:26:40 Creating measurements subscription for home xxx
Jan 08 17:26:40 vpngateway tibber-exporter[2336655]: 2024/01/08 17:26:40 Starting subscription xxx for home xxx
Jan 08 17:26:40 vpngateway tibber-exporter[2336655]: 2024/01/08 17:26:40 {"type":"connection_init","payload":{"token":"xxx"}} client
Jan 08 17:26:40 vpngateway tibber-exporter[2336655]: 2024/01/08 17:26:40 {"type":"connection_ack"} server
Jan 08 17:26:40 vpngateway tibber-exporter[2336655]: 2024/01/08 17:26:40 {"id":"xxx","type":"subscribe","payload":{"query":"subscription ($id:ID!){liveMeasurement(homeId: $id){timestamp,power,minPower,max>
Jan 08 17:26:40 vpngateway tibber-exporter[2336655]: 2024/01/08 17:26:40 Measurements subscription for home xxx connected
Jan 08 17:26:41 vpngateway tibber-exporter[2336655]: 2024/01/08 17:26:41 Starting http listener :8084
Jan 08 17:26:55 vpngateway tibber-exporter[2336655]: 2024/01/08 17:26:55 Measurements too old: 0001-01-01 00:00:00 +0000 UTC
Jan 08 17:27:25 vpngateway tibber-exporter[2336655]: 2024/01/08 17:27:25 Measurements too old: 0001-01-01 00:00:00 +0000 UTC
Jan 08 17:27:40 vpngateway tibber-exporter[2336655]: 2024/01/08 17:27:40 OnError: failed to read JSON message: failed to get reader: use of closed network connection
Jan 08 17:27:40 vpngateway tibber-exporter[2336655]: 2024/01/08 17:27:40 Exiting...

```

**Comment by terjesannum** [comment]
Tibber has implemented that change, and this exporter should be compatible with the updated service.
I see from my own logs that Tibber had some problems with their service today between 15:28 and 17:42. The exporter will recover automaticlly from such errors if you have automatic restart, have you checked it now or tried starting the exporter again?

**Comment by hlinden** [comment]
> Tibber has implemented that change, and this exporter should be compatible with the updated service. 

Oh, right, sorry then. I had only skimmed the error messages and seen "websocket" and thought _"well, that's it now"_ without further looking at it.

For me the last datapoint was at 16:27:45 and the next showed up at 18:46:30, both local time in Germany, so that matches your time.

>The exporter will recover automaticlly from such errors if you have automatic restart


Yes, I've got it running as a systemd service that will always restart the process, starting at 10 seconds recovery time and reaching 200 seconds after a few failures. That's probably why I saw my first datapoint about four minutes later than you did. 

Anyway, this is solved, thank you for your time, sorry, next time I'll do some more thorough debugging before a report.
